### PR TITLE
Add support for TDSType.GUID (fix) (#1582)

### PR DESCRIPTION
**Pull request by vxel**
This is a fix for the merged and reverted PR #2324.

The original PR intends to provide support for sending UUID parameters as `TDSType.GUID (0x24)` on the wire and (hopefully) implements feature request #1582. 
It currently requires the Java parameter be given to the driver as `String` with `JDBCType.GUID`, i.e. :
```
st.setObject(index, myUUID.toString(), microsoft.sql.Types.GUID);
```

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    @Test
+    public void testGuid() throws Exception {
+        try (Connection conn = getConnection(); Statement stmt = conn.createStatement()) {
+
+            // Create the test table
+            TestUtils.dropTableIfExists(escapedTableName, stmt);
+
+            String query = "create table " + escapedTableName
+                    + " (uuid uniqueidentifier, id int IDENTITY primary key)";
+            stmt.executeUpdate(query);
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java:52]: this test doesn't run in Azure SQL Data Warehouse

2024-03-31T21:30:07.8411163Z com.microsoft.sqlserver.jdbc.SQLServerException: Enforced unique constraints are not supported. To create an unenforced unique constraint you must include the NOT ENFORCED syntax as part of your statement.
2024-03-31T21:30:07.8412679Z 	at com.microsoft.sqlserver.jdbc.datatypes.GuidTest.testGuid(GuidTest.java:52)

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    @BeforeAll
+    public static void setupTests() throws Exception {
+        setConnection();
+    }
+
+    /*
+     * Test UUID conversions
+     */
+    @Test
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java:43]:
```suggestion
    @Test
    @Tag(Constants.xAzureSQLDW)
```

**Review by lilgreenbird — APPROVED** [review]

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
+package com.microsoft.sqlserver.jdbc.datatypes;
+
+import com.microsoft.sqlserver.jdbc.RandomUtil;
+import com.microsoft.sqlserver.jdbc.SQLServerResultSet;
+import com.microsoft.sqlserver.jdbc.TestResource;
+import com.microsoft.sqlserver.jdbc.TestUtils;
+import com.microsoft.sqlserver.testframework.AbstractSQLGenerator;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import microsoft.sql.Types;
+import org.junit.jupiter.api.BeforeAll;
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java:10]:
```suggestion
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Tag;
```

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+import java.sql.PreparedStatement;
+import java.sql.Statement;
+import java.util.UUID;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/GuidTest.java:26]:
```suggestion
import com.microsoft.sqlserver.testframework.Constants;
```

**Review by lilgreenbird — APPROVED** [review]

**Review by tkyc — APPROVED** [review]

**Comment by vxel** [comment]
@lilgreenbird Thanks for the correction. I wasn't aware of this Azure SQL Data Warehouse limitation.

**Review by Jeffery-Wasty — APPROVED** [review]

**Pull request merged**